magefile.go

```diff
@@ -639,10 +642,28 @@ func (Test) Unit() error {
 	}, "go", "test", "-cover", "-v", "-race", "./internal/...", "./cmd/...", "./pkg/...")
 }
 
-func (Test) Integration() error {
+func (Test) Integration(ctx context.Context) error {
+	workDir, ok := ctx.Value("workDir").(string)
+	if !ok || workDir == "" {
+		workDir = path.Join(cacheDir, "dev-env")
+	}
+	cluster, err := dev.NewCluster(
+		workDir,
+		dev.WithKubeconfigPath(os.Getenv("KUBECONFIG")), dev.WithSchemeBuilder(runtime.SchemeBuilder{operatorsv1alpha1.AddToScheme, aoapisv1alpha1.AddToScheme}),
+	)
+	if err != nil {
+		return fmt.Errorf("creating cluster client: %w", err)
+	}
+
+	if err := postClusterCreationFeatureToggleSetup(ctx, cluster); err != nil {
+		return fmt.Errorf("failed to perform post-cluster creation setup for the feature toggles: %w", err)
+	}
+	if err := deployFeatureToggles(ctx, cluster); err != nil {
+		return fmt.Errorf("failed to deploy feature toggles: %w", err)
+	}
```

Comment:
Do we need to wait for the addon-operator deployment to be ready here? Running `deployFeatureToggles` can potentially make ADO pod restart right?

Reply:
No i was saying we need to add a wait condition here.

Reply:
That wouldn't be necessary because when the integration test suite initiates, it waits on AddonOperator availability already

https://github.com/openshift/addon-operator/blob/main/integration/integration_test.go#L21

https://github.com/openshift/addon-operator/blob/main/integration/setup_test.go#L14-L38

Reply:
👍